### PR TITLE
feat(medusa): Support transformQuery/Body middleware

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
Move query transformation from the handler to a new pipe middleware

**How**
Provide a new pipe middleware to take care of those transformations and make it generic.
- Ensure consistent flow across the handlers
- Offer a single point of truth to transform the query
- Standardise the way the list/retrieve api works and allow users to use the standardise way as well as part of the DX

**Example**
Applied the new pipe middleware on some price lists end point as example

**Question**
With that pipe, do we still required the `controllers` to exists @srindom @olivermrbl ?

### CODE REVIEW

**Review by srindom — COMMENTED** [review]
Think this is super cool!

File: packages/medusa/src/api/middlewares/transformQuery.ts
Comment (adrien2p) [reply packages/medusa/src/api/middlewares/transformQuery.ts:1]: well catch :)

Comment (adrien2p) [reply packages/medusa/src/api/middlewares/transformQuery.ts]: @srindom let me know if I am wrong but at the moment the controllers does serve any other purpose than fetching simple data no ? since all the config is done before reaching the handler

**Comment by adrien2p** [comment]
Do not merge straight away

**Comment by adrien2p** [comment]
Before merging, just want to re check the typingd

**Comment by adrien2p** [comment]
@oli

**Review by zakariaelas — APPROVED** [review]
This is really cool 🤩 ! Super nice @adrien2p 🚀 

As for your comment, I personally think the current implementation is sufficient for now 👍

**Review by olivermrbl — COMMENTED** [review]
Really great work @adrien2p - will def. make our lives easier! 

Have added a single comment, but otherwise LGTM 👍

File: packages/medusa/src/utils/get-query-config.ts
Comment (adrien2p) [reply packages/medusa/src/utils/get-query-config.ts:112]: well spotted 👍 thanks

**Review by olivermrbl — APPROVED** [review]
Ready when requested change is included and conflicts are resolved 🚀

**Pull request merged**
